doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,484 @@
+processes. We could allow elevated windows to still create unelevated
+connections. However, we'd want to indicate per-pane what the elevation state
+of each connection is. The user would then need to keep track themselves of
+which terminal instances are elevated, and which are not.
+
+This also marks a departure from the current behavior, where everything in an
+elevated window would be elevated by default. The user would need to specify for
+each thing in the elevated window that they'd want to create it elevated. Or the
+Terminal would need to provide some setting like
+`"autoElevateEverythnigInAnElevatedWindow"`.
+
+We cannot support mixed elevation when starting in a unelevated window.
+Therefore, it doesn't make a lot of UX sense to support it in the other
+direction. It's a cleaner UX story to just have everything in a single window at
+the same elevation level.
+
+## Solution Design
+
+Instead of supporting mixed elevation in the same window, we'll introduce the
+following features to the Terminal. These are meant as a way of improving the
+quality of life for users who work in mixed-elevation (or even just elevated)
+environments.
+
+### Visible indicator for elevated windows
+
+As requested in #1939, it would be nice if it was easy to visibly identify if a
+Terminal window was elevated or not.
+
+One easy way of doing this is by adding a simple UAC shield to the left of the
+tabs for elevated windows. This shield could be configured by the theme (see
+[#3327]). We could provide the following states:
+* Colored (the default)
+* Monochrome
+* Hidden, to hide the shield even on elevated windows. This is the current
+  behavior.
+
+![UAC-shield-in-titlebar](UAC-shield-in-titlebar.png)
+_figure 1: a monochrome UAC shield in the titlebar of the window, courtesy of @mdtauk_
+
+### Change profile appearance for elevated windows
+
+In [#3062] and [#8345], we're planning on allowing users to set different
+appearances for a profile whether it's focused or not. We could do similar thing
+to enable a profile to have a different appearance when elevated. In the
+simplest case, this could allow the user to set `"background": "#ff0000"`. This
+would make a profile always appear to have a red background when in an elevated
+window.
+
+The more specific details of this implementation are left to the spec
+[Configuration object for profiles].
+
+### Configuring a profile to always run elevated
+
+Oftentimes, users might have a particular tool chain that only works when
+running elevated. In these scenarios, it would be convenient for the user to be
+able to identify that the profile should _always_ run elevated. That way, they
+could open the profile from the dropdown menu of an otherwise unelevated window
+and have the elevated window open with the profile automatically.
+
+We'll be adding the `"elevate": true|false` setting as a per-profile setting,
+with a default value of `false`. When set to `true`, we'll try to auto-elevate

```

Comment:
Should we maybe display the actual `command` before auto-elevate the profile? Malicious software could possibly modify the elevated profiles and perform some kind of injection. Users will not notice the difference if the profile is auto-elevated, and naturally say "yes" to some malwares unconsciously.

Reply:
I believe UAC will do that already, though, it's in the "show details". I suppose we'll only be showing the profile... I guess that means someone could edit the settings so the profile was `"commandline": "malicious.exe"`, and there's nothing preventing them from doing that. That's not good. Yea there should probably be another internal warning.